pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -184,6 +192,18 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable RequesterIdentit
       requestStatistics.setErrorCode(QueryException.PQL_PARSING_ERROR_CODE);
       return new BrokerResponseNative(QueryException.getException(QueryException.PQL_PARSING_ERROR, e));
     }
+    if (isLiteralOnlyQuery(brokerRequest)) {
```

Comment:
You need to move this block before the case insensitive handling (line 177) because that requires the table information. If table is not set (e.g. `SELECT now()`), that part will throw NPE.

Reply:
make sense, I will check table null in the case sensitive then direct return if it's null.